web/client/plugins/TOC/components/ArcGISLegend.jsx

```diff
@@ -31,7 +33,13 @@ function ArcGISLegend({
                     f: 'json'
                 }
             })
-                .then(({ data }) => setLegendData(data))
+                .then(({ data }) => {
+                    const dynamicLegendIsEmpty = data.layers.every(layer => layer.legend.length === 0);
+                    if ((node.dynamicLegendIsEmpty ?? null) !== dynamicLegendIsEmpty) {
+                        onUpdateNode({ dynamicLegendIsEmpty });
+                    }
```

Comment:
In https://github.com/allyoucanmap/MapStore2/commit/a11b6736dd0cc5175977ec75a212d3c7d205d309 the property as been renamed to `legendEmpty` to make it generic because it could be empty in other cases. Also the onUpdateNode has been changed to onChange to make it consistent with the parent component